units/event_metajob.go

```diff
@@ -106,15 +112,18 @@ func (j *eventMetaJob) dispatchLoop() error {
 			"event_type": j.events[i].ResourceType,
 		}))
 
-		// TODO: buffered writes after EVG-3062
-		if err = notification.InsertMany(notifications...); err != nil {
-			catcher.Add(err)
-			continue
+		for _, n := range notifications[0] {
+			catcher.Add(bulk.Append(n))
 		}
 
-		catcher.Add(j.dispatch(notifications))
-		catcher.Add(logger.MarkProcessed(&j.events[i]))
 	}
+	catcher.Add(bulk.Close())
+
+	for idx := range notifications {
+		catcher.Add(j.dispatch(notifications[idx]))
+		catcher.Add(logger.MarkProcessed(&j.events[idx]))
```

Comment:
I would prefer that you leave these where they were so that we don't collect too many at once, and so that we mark an event as processed as fast as possible in the event that we have a long running.

Is there any reason this needs be done related to buffered inserts?